Download Issues: Cancel download tasks when no episode matches

bjtitus opened the pull request:
On app-launch, this prunes download tasks which have no matching episode.

## To test

* Set a breakpoint at both of the cancels to see that they are run
* Kill the app by stopping in Xcode (force quitting will cause the tasks to be cancelled)
* Run the following shell script:
```bash
#!/bin/bash

SQL_COMMAND="DELETE FROM SJEpisode WHERE downloadTaskId IN (SELECT downloadTaskId FROM SJEpisode WHERE downloadTaskId IS NOT NULL LIMIT 10);"
sqlite3 "$(xcrun simctl get_app_container booted au.com.shiftyjelly.podcasts data)/Library/Application Support/Pocket Casts/podcast_newDB.sqlite3" "$SQL_COMMAND"

echo "Deleted first 10 episodes with downloadTaskId NOT NULL"
```
* Make sure breakpoints are hit
* Ensure episodes in podcast are no longer downloading
* Try downloading again and make sure they work

## Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `CHANGELOG.md` if necessary.
- [x] I have considered adding unit tests for my changes.
- [x] I have updated (or requested that someone edit) [the spreadsheet](https://docs.google.com/spreadsheets/d/107jqrutZhU0fVZJ19SBqxxVKbV2NWSdQC9MFYdLiAxc/edit?usp=sharing) to reflect any new or changed analytics.

bjtitus committed:
Cancel download tasks when no episode matches

# Conflicts:
#	podcasts/DownloadManager+SessionManagement.swift

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

bjtitus committed:
Skip episodeWithDownloadIds count for stuck downloads

# Conflicts:
#	podcasts/DownloadManager+SessionManagement.swift

bjtitus committed:
Refactor clearStuckDownloads

# Conflicts:
#	podcasts/DownloadManager+SessionManagement.swift
#	podcasts/DownloadManager.swift

bjtitus committed:
Add unit tests for clearStuckDownloads

# Conflicts:
#	PocketCastsTests/Extensions/FMDatabaseQueue+Test.swift

bjtitus committed:
Consolidate test code into DBTestCase helper

leandroalonso commented on podcasts/DownloadManager+SessionManagement.swift:
```diff
@@ -40,39 +40,34 @@ extension DownloadManager {
+                episodeUuids.remove(at: index)
+            } else {
+                if FeatureFlag.downloadFixes.enabled {
+                    task.cancel()
```
I think it would be good to add a log here.

bjtitus committed:
Merge branch 'trunk' into bjtitus/download-issues/cancel-untracked-downloads

# Conflicts:
#	podcasts/DownloadManager.swift
